### PR TITLE
Add support for resource requests

### PR DESCRIPTION
**Pull request by rouzwawi**
Example usage

```java
RunEnvironment environment =
    environment(
        "us.gcr.io/foo/bar:5",
        secret("gcp-key", "/etc/gcloud"))
    .withRequest(CPU.of("200m"))
    .withRequest(MEMORY.of("256Mi"));
```

### CODE REVIEW

File: hype-submitter/src/main/java/com/spotify/hype/model/CpuUnit.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+
+package com.spotify.hype.model;
+
+public enum CpuUnit {
```
Comment (rgruener) [inline hype-submitter/src/main/java/com/spotify/hype/model/CpuUnit.java]: is this enum still used?

File: hype-submitter/src/main/java/com/spotify/hype/model/MemoryUnit.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+
+package com.spotify.hype.model;
+
+public enum MemoryUnit {
```
Comment (rgruener) [inline hype-submitter/src/main/java/com/spotify/hype/model/MemoryUnit.java]: is this enum still used?

**Review by rgruener — CHANGES_REQUESTED** [review]
:+1: Looks good besides those unused classes

**Comment by rouzwawi** [comment]
@rgruener removed the unused enums

**Review by rgruener — APPROVED** [review]

**Pull request merged**